### PR TITLE
refactor(identity): fix import warnings in `keypair.rs`

### PR DESCRIPTION
**Pull request by tcoratger**
## Description

- Adding conditional `cfg` in `keypair.rs` imports.
- Adding `#[allow(unused_variables)]` on the top of the `sign` function.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by mxinden — DISMISSED** [review]
That makes sense to me. Thank you @tcoratger for your continuous help on rust-libp2p! :heart:

**Comment by mxinden** [comment]
@tcoratger I don't think there is any need to merge `master` after the `send-it` label has been applied. Mergify will merge `master` into this pull request once it is first in the merge queue.

**Review by mxinden — APPROVED** [review]

File: identity/src/keypair.rs [hunk 1]
```diff
@@ -27,7 +33,19 @@ use crate::error::{DecodingError, SigningError};
...
+    feature = "rsa"
+))]
 use quick_protobuf::{BytesReader, Writer};
+#[cfg(any(
+    feature = "ecdsa",
+    feature = "secp256k1",
+    feature = "ed25519",
+    feature = "rsa"
+))]
 use std::convert::TryFrom;
```
Comment (thomaseizinger) [inline identity/src/keypair.rs:49]: Do we even need to import this? I thought that is part of the prelude starting with the 2018 or 2021 edition?

File: identity/src/keypair.rs [hunk 2]
```diff
@@ -18,6 +18,12 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
+#[cfg(any(
+    feature = "ecdsa",
+    feature = "secp256k1",
+    feature = "ed25519",
+    feature = "rsa"
+))]
 use crate::error::OtherVariantError;
```
Comment (thomaseizinger) [inline identity/src/keypair.rs:27]: Alternatively, we could use full-qualified paths at the usage site.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**